pl_bolts/metrics/object_detection.py

```diff
@@ -2,7 +2,16 @@
 from torch import Tensor
 
 
-def iou(preds: Tensor, target: Tensor) -> Tensor:
+def _evaluate_iou(preds: torch.Tensor, target: torch.Tensor):
+    """Evaluate intersection over union (IOU) for target from dataset and output prediction from model."""
+
+    if preds["boxes"].shape[0] == 0:
+        # no box detected, 0 IOU
+        return torch.tensor(0.0, device=preds["boxes"].device)
+    return iou(target["boxes"], preds["boxes"]).diag().mean()
+
+
+def iou(preds: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
```

Comment:
can we use rather a TorchMetrics version?

Reply:
Sorry for the delay. I wasn't well past week.
I could :( But it would take more time for me as I haven't used torchmetrics up lately.

Reply:
I don't think we have a iou metric for detection in torchmetrics, so let's keep as is for now.

Reply:
FYI, RetinaNet now uses `torchvision.ops.box_iou` instead of our in-house `iou` here.